File: Actividad 2/C#/actividad1.cs
// para ejecutar mcs -out:actividad1.exe actividad1.cs 
// Angel David Veron Cuevas
using System;
using System.Collections.Generic;

namespace Project_1 {
    class MainClass {
        public static void Main (string[] args) {
            Console.WriteLine("Ingresa una cadena:");
            String cadena;
            cadena=Console.ReadLine();
            Console.WriteLine("El texto introducido es: " + cadena);
            bool espacio;
            Dictionary<string, int> letrasRepetidas = new Dictionary<string, int>();  
            int value;
            bool hasValue;
            for (int i = 0; i < cadena.Length; i++){
                hasValue = letrasRepetidas.TryGetValue(cadena.Substring(i, 1), out value);
                espacio = (Char.IsWhiteSpace(cadena, i));
                if (hasValue)
                    letrasRepetidas[cadena.Substring(i, 1)] = letrasRepetidas[cadena.Substring(i, 1)] +  1;
                else if (!espacio)
                    letrasRepetidas.Add(cadena.Substring(i, 1), 1);  
            }
            foreach (KeyValuePair<string, int> letra in letrasRepetidas)
            {
                Console.WriteLine("La letra :"+ letra.Key + " se repite  " + letra.Value );
            }
        }
    }
}

File: Actividad 2/Java/actividad1.java
//Angel David Veron Cuevas
import java.util.HashMap; // import the HashMap class
import java.util.Scanner;

public class actividad1 {

  public static void main(String[] args) {
    HashMap<Character, Integer> letras_repetidas = new HashMap<Character, Integer>();
    Scanner scann = new Scanner(System.in);
    String cadena = scann.nextLine();
    Integer contador = 0;
    for (int i = 0; i < cadena.length(); i++) {
      if (cadena.charAt(i) != ' ') {
        for (Character key : letras_repetidas.keySet()) {
          if (key == cadena.charAt(i)) {
            contador = letras_repetidas.get(cadena.charAt(i));
            contador += 1;
            letras_repetidas.replace(cadena.charAt(i), contador);
          }
        }
        
        if (contador == 0) {
          letras_repetidas.put(cadena.charAt(i), 1);
        }
        contador = 0;
      }
    }
     System.out.println(letras_repetidas);
  }
}
